platform/fabric/core/generic/chaincode/discovery.go

```diff
@@ -93,9 +93,29 @@ func (d *Discovery) Call() ([]view.Identity, error) {
 	if err != nil {
 		return nil, errors.WithMessagef(err, "failed getting endorsers for [%s]", key)
 	}
-	var endorserIdentities []view.Identity
-	for _, e := range endorsers {
-		endorserIdentities = append(endorserIdentities, e.Identity)
+	configResult, err := cr.Config()
+	if err != nil {
+		return nil, errors.WithMessagef(err, "failed getting config for [%s]", key)
+	}
+
+	var discoveredEndorsers []driver.DiscoveredPeer
+	for _, peer := range endorsers {
+		// extract peer info
+		member := peer.AliveMessage.GetAliveMsg().Membership
```

Comment:
GetAliveMessage can be nil

Reply:
Under which conditions?

Reply:
Well you receive this from a peer, what if the peer is faulty and doesn't put the message there? 
Yes, continue to other peers.

Reply:
all right